### DISCUSSION TITLE
Efficient way to Run Model For Multiple Policies

### DISCUSSION THREAD

**Discussion by mbici** [fumitoh/modelx#145] [General]
I have created a model that works well for individual policy reserve calculations, I now want to run the model acrosss multiple policies. How do I go about , I need to store all the cashflows at each time e.g premiums, benefits for each of the policies

**Comment by fumitoh** [comment]
A common approach would involve the following steps:

* Create a cells that reads in the entire model point table as a pandas DataFrame indexed with policy ID.
* Parameterize your space by policy ID.
* Make policy attributes and assumptions read values from the table by looking up the parameter.
* Individual policy results are created as parameterized spaces, such as `Projection[1]` for Policy ID 1, `Projection[2]` for Policy 2, and so on.

Here are simplified examples for the steps above:

```python
@mx.defcells
def model_point_table():
    # Assuming model_point_table.csv is in the same directory as the model.
    return pd.read_csv(_model.path.parent / "model_point_table.csv")   
```

```python
space.parameters = ("policy_id",)    # Assuming your space is assigned to 'space'.
```

```python
@mx.defcells
def issue_age():
    return model_point_table().loc[policy_id, "issue_age"]
```

```python
print(space[3].premiums(10))    # Get the premiums of Policy 3 at time 10 and print it out.
```

 I think examining simple models in lifelib, such as BasicTerm_S in the basiclife library is helpful.
